firebase-firestore/src/main/java/com/google/firebase/firestore/core/Filter.java

```diff
@@ -15,11 +15,19 @@
 package com.google.firebase.firestore.core;
 
 import com.google.firebase.firestore.model.Document;
+import com.google.firebase.firestore.model.FieldPath;
+import java.util.List;
 
 public abstract class Filter {
   /** Returns true if a document matches the filter. */
   public abstract boolean matches(Document doc);
 
   /** A unique ID identifying the filter; used when serializing queries. */
   public abstract String getCanonicalId();
+
+  /** Returns a list of all field filters that are contained within this filter */
+  public abstract List<FieldFilter> getAllFieldFilters();
```

Comment:
It's not very obvious to me what the difference between "getFilters" in CompositeFIlter and and "getAllFieldFilters" is. Should it be getFIlters/getFlattenedFilters. As it stands now, getAllFieldFilters() suggests that it is is merely `getFilters().filter(f -> f instanceof FieldFilter)`.